src/View/SSViewer.php

```diff
@@ -336,288 +193,117 @@ public static function get_templates_by_class($classOrObject, $suffix = '', $bas
     }
 
     /**
-     * Get the current item being processed
+     * Get an associative array of names to information about callable template provider methods.
      *
-     * @return ModelData
-     * @deprecated 5.4.0 Will be removed without equivalent functionality to replace it.
+     * @var boolean $createObject If true, methods will be called on instantiated objects rather than statically on the class.
      */
-    public static function topLevel()
+    public static function getMethodsFromProvider(string $providerInterface, $methodName, bool $createObject = false): array
```

Comment:
This used to be `SSViewer_DataPresenter::getPropertiesFromProvider()`.

Moved here because it's generic, and can be used by other template engines. Renamed because they're methods, not properties (even if they can be invoked _as though they were_ properties).